service/s3/s3crypto/s3crypto_encryption_client.go

```diff
@@ -75,14 +82,29 @@ func NewEncryptionClient(prov client.ConfigProvider, builder ContentCipherBuilde
 //	})
 //	err := req.Send()
 func (c *EncryptionClient) PutObjectRequest(input *s3.PutObjectInput) (*request.Request, *s3.PutObjectOutput) {
-	req, out := c.S3API.PutObjectRequest(input)
+	req, out := c.S3Client.PutObjectRequest(input)
 
-	// Create temp file to be used later for calculating the SHA256 header
-	f, err := ioutil.TempFile(c.Config.TempFolderPath, "")
+	var dst io.ReadWriteSeeker
+
+	// Get Size of file
+	n, err := input.Body.Seek(0, 2)
 	if err != nil {
 		req.Error = err
 		return req, out
 	}
+	input.Body.Seek(0, 0)
+
+	useTempFile := n > c.Config.MinFileSize
+	if useTempFile {
+		// Create temp file to be used later for calculating the SHA256 header
+		dst, err = ioutil.TempFile(c.Config.TempFolderPath, "")
+		if err != nil {
+			req.Error = err
+			return req, out
+		}
+	} else {
+		dst = &bytesWriteSeeker{}
+	}
```

Comment:
I suggest moving all of the temp file logic into a helper function. This will simplify reading this function.  In addition if the in memory buffer satisfies the same `ReadWriteSeekCloser` interface then special case logic is not needed at line 147